src/manager.ts

```diff
@@ -184,16 +211,118 @@ export class SecretsManager implements ISecretsManager {
     this._attachedInputs.delete(attachedId);
   }
 
-  private _connector: ISecretsConnector;
+  private _check(token: symbol, namespace: string): void {
+    if (Private.isLocked() || Private.namespace.get(token) !== namespace) {
+      throw new Error(
+        `The secrets namespace ${namespace} is not available with the provided token`
+      );
```

Comment:
```suggestion
      Private.lock(
        `The secrets namespace ${namespace} is not available with the provided token`
      );
```

Reply:
Don't know if we want to lock the manager on any failing attempt. It should not occurs, though.

Reply:
I don't feel strongly about this, but the only times I expect this to happen are in development so I erred on the side of security. I'm happy to defer to your judgement.

Reply:
You could use a `strict` flag that can be set when starting jupyterlab.